### PR TITLE
Refactor cpp symbols detection.

### PR DESCRIPTION
**Pull request by devnexen**
Merging them into one and returning similar value.
"Fix" for #11662

### CODE REVIEW

**Comment by radare** [comment]
Dont think this is a c++ bin but i didnt checked

File: libr/bin/p/bin_elf.inc [hunk 1]
```diff
@@ -1118,7 +1102,7 @@ static RBinInfo* info(RBinFile *bf) {
 	if (!(ret = R_NEW0 (RBinInfo))) {
 		return NULL;
 	}
-	ret->lang = is_cpp (bf) ? "c++" : "c";
+	ret->lang = r_bin_is_cxx (bf) ? "cxx" : "c";
```
Comment (ret2libc) [inline libr/bin/p/bin_elf.inc]: why did you change the string from `c++` to `cxx`? I'd keep it as it was if there is no particular reason.

Comment (devnexen) [reply libr/bin/p/bin_elf.inc]: It s to be "in line" with the mangling feature change. But I can put it back if it triggers annoyance.

Comment (ret2libc) [reply libr/bin/p/bin_elf.inc]: I see

File: libr/include/r_bin.h [hunk 2]
```diff
@@ -820,6 +820,7 @@ R_API void r_bin_filter_sections (RList *list);
 R_API void r_bin_filter_classes (RList *list);
 R_API bool r_bin_strpurge(RBin *bin, const char *str, ut64 addr);
 R_API bool r_bin_string_filter(RBin *bin, const char *str, ut64 addr);
+R_API bool r_bin_is_cxx (RBinFile *binfile);
```
Comment (ret2libc) [inline libr/include/r_bin.h]: style: no space before `(`.

File: libr/bin/p/bin_elf.inc (hunk 1 above)
Comment (ret2libc) [inline libr/bin/p/bin_elf.inc]: I think this change actually changes the semantic of the check... before we were just checking imports, now you are checking all symbols, right? Is that what you wanted?

Also, shall we consider C binaries compiled with ASAN (and that for this reasons have some mangled names) C++ binaries? Or shall we filter those ASAN symbols out?

Comment (devnexen) [reply libr/bin/p/bin_elf.inc]: No after second thoughts I prefer not considering C sanitized binaries be seen as.

**Comment by radare** [comment]
Merging, but at some point we must improve the language detectors to avoid performance costs when loading bins with lot of symbols
